synfig-core/src/modules/mod_ffmpeg/trgt_ffmpeg.cpp

```diff
@@ -195,49 +182,40 @@ ffmpeg_trgt::init(ProgressCallback *cb=NULL)
 	}
 
 	String ffmpeg_binary_path;
-	std::list< String > binary_choices;
-	binary_choices.push_back("ffmpeg");
-	binary_choices.push_back("avconv");
-	std::list< String >::iterator iter;
-	for(iter=binary_choices.begin();iter!=binary_choices.end();iter++)
-	{
-		String binary_path;
 #if defined(WIN32_PIPE_TO_PROCESSES)
-		binary_path = synfig::get_binary_path("");
-		if (binary_path != "")
-			binary_path = etl::dirname(binary_path)+ETL_DIRECTORY_SEPARATOR;
-		binary_path += *iter+".exe";
-		if( access( binary_path.c_str(), F_OK ) != -1 ) {
-			binary_path = "\"" + binary_path + "\"";
-			ffmpeg_binary_path = binary_path;
-			break;
-		}
+	// Windows always have ffmpeg
+	std::string binary_path = etl::dirname(synfig::get_binary_path(".")) + "/ffmpeg.exe";
+	if (g_access(binary_path.c_str(), F_OK ) != -1 ) { // File found
+		ffmpeg_binary_path = "\"" + binary_path + "\"";
+	}
 #else
-		binary_path = *iter;
-		
-		String command;
+	// Some Linux OS may have `avconv` instead of `ffmpeg`, so let's check both
+	const std::vector<std::string> binary_choices = {"ffmpeg", "avconv"};
+	for (const auto& bin_name : binary_choices) {
+		std::string command = "which " + bin_name;
 		String result;
-		command = "which "+binary_path;
 		FILE* pipe = popen(command.c_str(), "r");
 		if (!pipe) {
 			continue;
 		}
-		char buffer[128];
+		char buf[128];
 		while(!feof(pipe)) {
-			if(fgets(buffer, 128, pipe) != NULL)
-					result += buffer;
+			if(fgets(buf, 128, pipe) != NULL)
+				result += buf;
 		}
-		pclose(pipe);
-		synfig::info(strprintf("\"%s\" --> %s", command.c_str(), result.c_str()));
-		if (result.substr(0,1) == "/" || result.substr(0,2) == "./" ){
-			ffmpeg_binary_path = binary_path;
+		/* `which` exit status
+		 * 0      if all specified commands are found and executable
+		 * 1      if one or more specified commands is nonexistent or not executable
+		 * 2      if an invalid option is specified */
+		synfig::info(etl::strprintf("\"%s\" --> %s", command.c_str(), result.c_str()));
```

Comment:
no need to use `strprintf` here: `synfig::info()` accepts format + values.

Reply:
fixed